### PR TITLE
guides: add pushToAlgoliaWeb script

### PR DESCRIPTION
**Pull request by levimichael**
## 🧭 What and Why

🎟 JIRA Ticket: SELFSER-1038

### Changes included:

- Adds a script to push code snippets to https://github.com/algolia/AlgoliaWeb

### CODE REVIEW

File: scripts/ci/codegen/pushToAlgoliaWeb.ts [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+  OWNER,
+  run,
+  setVerbose,
+  toAbsolutePath,
+} from '../../common.js';
+import { getNbGitDiff } from '../utils.js';
+
+import { commitStartRelease } from './text.js';
+
+const languageFiles = {
```
Comment (shortcuts) [inline scripts/ci/codegen/pushToAlgoliaWeb.ts:20]: Can we make this programmatic instead? We have clients.config.json file that contains every available langue and their snippet location, it should be enough to build the path and avoid us maintaining the file list

File: .github/workflows/push-to-algolia-web.yml [hunk 2]
```diff
@@ -0,0 +1,24 @@
+name: Push snippets to AlgoliaWeb
+
+on: workflow_dispatch
+
+jobs:
+  release:
+    name: Scheduled Release
```
Comment (shortcuts) [inline .github/workflows/push-to-algolia-web.yml:7]:
```suggestion
  push:
    name: Manual trigger push for onboarding guides
```

File: scripts/ci/codegen/pushToAlgoliaWeb.ts [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+
+  if ((await getNbGitDiff({ head: null, cwd: tempGitDir })) === 0) {
+    console.log('❎ Skipping push to AlgoliaWeb because there is no change.');
+
+    return;
+  }
+
+  await configureGitHubAuthor(tempGitDir);
+
+  const message = 'feat: update specs and supported versions';
```
Comment (shortcuts) [inline scripts/ci/codegen/pushToAlgoliaWeb.ts:90]: This should be something related to guides I believe

File: scripts/ci/codegen/pushToAlgoliaWeb.ts [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+
+  console.log(`Creating pull request on ${OWNER}/${repository}...`);
+  const octokit = getOctokit();
+  const { data } = await octokit.pulls.create({
+    owner: OWNER,
+    repo: repository,
+    title: message,
+    body: [
+      'This PR is automatically created by https://github.com/algolia/api-clients-automation',
+      'It contains the latest generated code snippets.',
```
Comment (shortcuts) [inline scripts/ci/codegen/pushToAlgoliaWeb.ts:107]:
```suggestion
      'It contains the latest generated guides.',
```

File: scripts/ci/codegen/pushToAlgoliaWeb.ts [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+    ].join('\n\n'),
+    base: 'master',
```
Comment (shortcuts) [inline scripts/ci/codegen/pushToAlgoliaWeb.ts:109]: Shouldn't we target dev for algoliaWeb?

**Review by shortcuts — COMMENTED** [review]
🤩

File: scripts/ci/codegen/pushToAlgoliaWeb.ts (hunk 1 above)
Comment (levimichael) [reply scripts/ci/codegen/pushToAlgoliaWeb.ts:20]: Ah! Yes indeed!

File: scripts/ci/codegen/pushToAlgoliaWeb.ts (hunk 3 above)
Comment (levimichael) [reply scripts/ci/codegen/pushToAlgoliaWeb.ts:90]: Good catch, thank you!

File: scripts/ci/codegen/pushToAlgoliaWeb.ts (hunk 5 above)
Comment (levimichael) [reply scripts/ci/codegen/pushToAlgoliaWeb.ts:109]: Yes, indeed 😅

**Review by shortcuts — APPROVED** [review]
let's try this!

**Pull request merged**